### PR TITLE
Update tqdm.py

### PR DESCRIPTION
**Pull request by formacube**
add possibility to cumulate or not progress to the parent task progress if parent_task_id is provided

### CODE REVIEW

**Comment by jedie** [comment]
Thanks for contribution...

But why do we need `cumulate_w_parent_progress` ???
You can pass `parent_task_id` or not to update the main task or not.

So i assume this is double code for the same, isn't it?

Expand the README is fine... On the other hand, this has a general problem: it is not guaranteed that the code works, because it is untested. Actually, it is better to add example code and refer to it. Because they can be added to the tests.

Maybe we should enhance https://github.com/boxine/django-huey-monitor/blob/master/huey_monitor_tests/test_app/tasks.py and add more DocStrings there and make more links from README to it?

**Comment by formacube** [comment]
I implemented it this way with `cumulate_w_parent_progress=True` as default parameter so that it is directly compatible with all code written so far by the current users.

For me, from a conceptual point of view, we should have `cumulate_w_parent_progress=False` as default, as there is no reason why a sub-task counter is in the same order of magnitude as the main-task counter, and why if progress from the subtask is increased by 1, progress of main task will be increased by 1.


In my different types of application, the progress counters of the main task correspond to the number of sub-tasks to be run.
So each time a sub-task is completed, main task-counter has to be increased by 1.

But on the other side, as I told you when I opened the issue, to be useful, the `update_dt` of the main task has to be updated each time a sub-task updates its progress.

Essentially because a sub-task can take 10h or 24h to complete.

Last month, I had a process which was not showing any progress for 3 weeks.
I had no clue what was going on and as those processes can take a lot of time, I was hesitant to stop and relaunch it.

It made me decide to invest more effort into designing a more efficient progress monitoring of my tasks.


For the testing, I never had professional training on it and I don't know how you work, so anything is fine by me, and I'll be glad to learn how you handle it and integrate your best practices into the ways I work.

File: huey_monitor/tqdm.py [hunk 1]
```diff
@@ -21,7 +21,8 @@ def __init__(self,
                  total=None,
                  unit='it',
                  unit_divisor=1000,
-                 parent_task_id=None
+                 parent_task_id=None,
+                 cumulate_w_parent_progress=True,
```
Comment (jedie) [inline huey_monitor/tqdm.py:25]: Can you rename `cumulate_w_parent_progress` to something shorter? e.g.: `cumulate2parents` ?

File: README.md [hunk 2]
```diff
@@ -73,6 +73,43 @@ def foobar_task(task, list_to_process):
...
+        foobar_subtask(item, parent_task_id = task.id)
+        process_info.update(n=1) # add the information that one item was processed
+
+@huey.db_task(context=True)
+def foobar_subtask(item, task=None, parent_task_id=None):
+    process_info = ProcessInfo(
+        task,
+        parent_task_id=parent_task_id, # to link progress reporting to the parent task
+        
+        cumulate_w_parent_progress=False, # optional: wether or not  the progress reported on the sub task will be cumulated also at the parent task level
```
Comment (jedie) [inline README.md:96]: This line seems to be too long, isn't it?

Comment (formacube) [reply README.md]: OK, I wrapped it and also change  the reference to `cumulate_w_parent_progress` into `cumulate2parents`

**Comment by formacube** [comment]
I implemented your comments.
it should be ok now

File: huey_monitor/tqdm.py [hunk 3]
```diff
@@ -80,13 +83,16 @@ def update(self, n=1):
         if self.parent_task_id:
             # Store information for main task, too:
             ids.append(self.parent_task_id)
-            objects.append(
-                TaskProgressModel(
-                    task_id=self.parent_task_id,
-                    progress_count=n,
-                    create_dt=now
+            
+            if self.cumulate_w_parent_progress:
```
Comment (jedie) [inline huey_monitor/tqdm.py:87]: this is missing

Comment (formacube) [reply huey_monitor/tqdm.py]: ok, solved

**Comment by jedie** [comment]
@formacube linting failes, see: https://github.com/boxine/django-huey-monitor/runs/4324397296?check_suite_focus=true#step:9:24

You can run tests and linting local with:
```
make pytest
make lint
```
Helpfull is also: `make fox-code-style` ;)

**Comment by formacube** [comment]
linting should be ok and in line with your personal perferences.

I use double line breaks to make code reading & structure easier. That's my preference.
But you are the code owner, so just put it to your preference.

No need to involve me in the future if you prefer lines of 119 characters or if you don't like 2 lines break in a row.

**Review by formacube — COMMENTED** [review]
ok for me

**Comment by formacube** [comment]
Can you go on with this ?
I did the cosmetic changes you requested 1 month ago

**Review by formacube — COMMENTED** [review]
ok, fixed 2 other whitespaces ...
hop I will not spend my life there chasing whitespaces !

**Comment by jedie** [comment]
It's green... But i think about to refactor this code part, see: https://github.com/boxine/django-huey-monitor/discussions/57

So this PR is useless :(

**Comment by jedie** [comment]
I close this in favour of a complete rewrite here: https://github.com/boxine/django-huey-monitor/pull/58

**Comment by jedie** [comment]
Implement the idea with https://github.com/boxine/django-huey-monitor/pull/61

**Pull request closed without merging**